### PR TITLE
Update the devfile samples fetching and some css tweaks

### PR DESCRIPTION
**Pull request by rohitkrai03**
## Fixes
- https://issues.redhat.com/browse/HAC-462.

## Description

- Because of CORS issue in the devfile registry API, we were using mock data for devfile samples. https://github.com/devfile/api/issues/719 is fixed now. 
- Removed mock data and added logic to fetch devfile samples from real API.
- Updated some styles to match the UX designs accurately.
- Updated the model and usage of `ComponentDetectionQuery`.

## Type of change

- [x] Feature

## Screen shots / Gifs for design review 

https://user-images.githubusercontent.com/6041994/150354183-4673edbb-6bab-4159-9c2c-cc62841edaa6.mov

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/ReviewComponents/ReviewComponentsForm.tsx [hunk 1]
```diff
@@ -23,16 +23,18 @@ export const ReviewComponentsForm: React.FC<ReviewComponentsFormProps> = ({
...
+            isInline
+            name="deployMethod"
+            options={[
+              { label: 'Automatically deploy', value: DeployMethod.AutomaticDeploy },
+              { label: 'Manually deploy', value: DeployMethod.ManualDeploy },
+            ]}
+          />
+        </div>
       )}
-      <PageSection className="hacDev-review-form__components-section">
```
Comment (rottencandy) [inline src/components/ReviewComponents/ReviewComponentsForm.tsx:35]: This is much better. I think we can also delete the unused class in ReviewComponentsForm.scss ?

File: src/utils/__tests__/devfile-utils.spec.ts [hunk 2]
```diff
@@ -1,4 +1,83 @@
-import { mockItems, getDevfileSamples } from '../devfile-utils';
+import { getDevfileSamples, DevfileSample } from '../devfile-utils';
+
+/**
+ * TODO: remove mock data for samples
+ */
```
Comment (rottencandy) [inline src/utils/__tests__/devfile-utils.spec.ts:5]: Tiny nit: The TODO text won't be needed anymore I guess

File: src/utils/devfile-utils.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import { CatalogItem } from './../shared/components/catalog/utils/types';
+// import { CatalogItem } from './../shared/components/catalog/utils/types';
+// import { coFetchJSON } from './../dynamic-plugin-sdk';
```
Comment (rottencandy) [inline src/utils/devfile-utils.ts:2]: Nit: remove commented code

**Review by rottencandy — COMMENTED** [review]
Thanks! Tested locally and everything looks good. Just had a few very minor suggestions.

File: src/components/ReviewComponents/ReviewComponentsForm.tsx (hunk 1 above)
Comment (rohitkrai03) [reply src/components/ReviewComponents/ReviewComponentsForm.tsx:35]: Done. Removed the unused css file.

File: src/utils/__tests__/devfile-utils.spec.ts (hunk 2 above)
Comment (rohitkrai03) [reply src/utils/__tests__/devfile-utils.spec.ts]: Removed

File: src/utils/devfile-utils.ts (hunk 3 above)
Comment (rohitkrai03) [reply src/utils/devfile-utils.ts]: Removed

**Review by rottencandy — APPROVED** [review]
/lgtm

**Pull request merged**